src/main/java/org/springframework/data/redis/connection/jedis/JedisConnectionFactory.java

```diff
@@ -667,32 +665,29 @@ public void start() {
 			if (isRedisClusterAware()) {
 
 				this.cluster = createCluster(getClusterConfiguration(), getPoolConfig());
-				this.topologyProvider = createTopologyProvider(this.cluster);
-				this.clusterCommandExecutor = new ClusterCommandExecutor(this.topologyProvider,
-						new JedisClusterConnection.JedisClusterNodeResourceProvider(this.cluster, this.topologyProvider),
-						EXCEPTION_TRANSLATION, executor);
+				this.topologyProvider = createClusterTopologyProvider(this.cluster);
+				this.clusterCommandExecutor = createClusterCommandExecutor(this.cluster, this.topologyProvider);
 			}
 
 			this.state.set(State.STARTED);
 		}
 	}
 
-	private boolean isCreatedOrStopped(@Nullable State state) {
-		return State.CREATED.equals(state) || State.STOPPED.equals(state);
-	}
-
 	@Override
 	public void stop() {
 
 		if (this.state.compareAndSet(State.STARTED, State.STOPPING)) {
 
 			if (getUsePool() && !isRedisClusterAware()) {
-				if (this.pool != null) {
+
+				Pool<Jedis> jedisPool = this.pool;
+
+				if (jedisPool != null) {
 					try {
-						this.pool.close();
+						jedisPool.close();
 						this.pool = null;
-					} catch (Exception ex) {
-						log.warn("Cannot properly close Jedis pool", ex);
+					} catch (Exception cause) {
```

Comment:
To be precise: `cause` isn't true as the cause refers to the exception cause and not the caught exception. Please stick to `ex`.

Reply:
`cause` is the reason the error occurred in the block above. You are confusing this with caused by.

Reply:
In any case, I am not going to argue about this.

If you want it to be consistent, then we should call it `e` everywhere Exceptions are caught (or `ignore` when Exceptions are not handled), which is what the core Spring Framework uses (i.e. `e`, not sure about `ignore`).

Using `ex` is not consistent, and `o_O` is not professional.